src/isolate.rs

```diff
@@ -254,34 +266,156 @@ impl Isolate {
     Ok(())
   }
 
+  pub fn mod_new(
+    &mut self,
+    name: String,
+    source: String,
+  ) -> Result<libdeno::deno_mod, JSError> {
+    let name_ = CString::new(name.clone()).unwrap();
+    let name_ptr = name_.as_ptr() as *const i8;
+
+    let source_ = CString::new(source.clone()).unwrap();
+    let source_ptr = source_.as_ptr() as *const i8;
+
+    let id = unsafe {
+      libdeno::deno_mod_new(self.libdeno_isolate, name_ptr, source_ptr)
```

Comment:
Are you sure name_ptr and source_ptr are copied by deno_mod_new? It looks like they might not, in which case this is unsafe, since name_ and source_ get dropped at the end of this scope.

Reply:
Yes, because of the above clones.

Reply:
That's not an acceptable answer. The clones also get dropped when they go out of scope.

Reply:
Oh - deno_mod_new does not take ownership of its parameters - those strings are copied and internalized into v8